docs/site/guide/overview.md

```diff
@@ -0,0 +1,134 @@
+and type classes) into the forefront.  Instead of strongly encouraging developers to rewrite all 
+their code in Swift, it pragmatically focuses on interoperability with other languages, e.g., 
+allowing you to directly import C header files and use them without an 
+[FFI](https://en.wikipedia.org/wiki/Foreign_function_interface) and (now) the ability to use Python 
+APIs without wrappers.
+
+Swift has the audacious goal of spanning all the way from low-level systems programming to 
+high-level scripting, with a focus on being
+[easy to learn and use](https://www.apple.com/swift/playgrounds/). 
+Because Swift needs to be easy to learn and use but also powerful, it relies on the principle of 
+[progressive disclosure of complexity](https://www.nngroup.com/articles/progressive-disclosure/), 
+which aggressively factors the cost of complexity onto the people who benefit from that complexity. 
+The "scripting language feel" combined with high performance is very useful for machine learning.
+
+A final pertinent aspect of the design of Swift is that much of the Swift language is actually 
+implemented in its standard library.  "Builtin" types like 
+[Int](https://developer.apple.com/documentation/swift/int) and 
+[Bool](https://developer.apple.com/documentation/swift/bool) are actually just structs defined in 
+the standard library that wrap magic types and operations.  As such, sometimes we joke that Swift 
+is just "syntactic sugar for LLVM".
+
+There is a lot more that is cool about Swift and a ton of content available online.  If you are 
+interested in learning more about general Swift programming concepts, here are a few links to get 
+started:
+
+ - [A Swift Tour](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/GuidedTour.html) is a skimmable tour of the high level syntax and feel of Swift, and is part of the larger "The Swift Programming Language" book.
+ - Value semantics are powerful and play an important role in Swift code, as explained in "[Building Better Apps with Value Types in Swift](https://developer.apple.com/videos/play/wwdc2015/414/)" [[YouTube](https://www.youtube.com/watch?v=av4i3x-aZbM)].
+ - Swift supports classic OOP, but has adapted ideas from the Haskell type system.  This is explained in "[Protocol-Oriented Programming in Swift](https://developer.apple.com/videos/play/wwdc2015/408/)" [[YouTube](https://www.youtube.com/watch?v=g2LwFZatfTI)].
+
+One warning: Swift evolved rapidly in its early years, so you should be careful with anything 
+before Swift 3 (released in 2016).
+
+## Why Swift for TensorFlow?
+
+Swift for TensorFlow is a new way to develop machine learning models. It
+gives you the power of
+[TensorFlow](https://www.tensorflow.org) directly integrated into the
+[Swift programming language](https://swift.org/about). We believe that
+machine learning paradigms are so important that they deserve
+**first-class language and compiler support**.
```

Comment:
Also, maybe it makes sense to have the _Why S4TF_ section come before _Why Swift_. The former (this section) mentions that the project has first-class lang and compiler support + autodiff <- what a lot of folks in research care about. 

I think not many people read the whole post/article/guide when they scan text for the first time, so having a section on the S4TF system in the beginning may make more sense.